versioning/BRM/extentmap.cpp

```diff
@@ -219,15 +219,20 @@ boost::mutex ExtentMapRBTreeImpl::fInstanceMutex;
 ExtentMapRBTreeImpl* ExtentMapRBTreeImpl::fInstance = nullptr;
 
 /*static*/
-ExtentMapRBTreeImpl* ExtentMapRBTreeImpl::makeExtentMapRBTreeImpl(unsigned key, off_t size, bool readOnly)
+ExtentMapRBTreeImpl* ExtentMapRBTreeImpl::makeExtentMapRBTreeImpl(unsigned key, off_t size, bool readOnly,
+                                                                  const MasterSegmentTable* emSegTable)
 {
   boost::mutex::scoped_lock lk(fInstanceMutex);
 
   if (fInstance)
   {
     if (key != fInstance->fManagedShm.key())
     {
+      if (emSegTable)
+        emSegTable->getTable_upgrade(MasterSegmentTable::EMTable);
       fInstance->fManagedShm.reMapSegment();
+      if (emSegTable)
+        emSegTable->getTable_downgrade(MasterSegmentTable::EMTable);
```

Comment:
This should be changed to:
```c++
      if (emSegTable)
      {
        emSegTable->getTable_upgrade(MasterSegmentTable::EMTable);
        emLocked = true;
      }
      fInstance->fManagedShm.reMapSegment();
      if (emSegTable)
      {
        emLocked = false;
        emSegTable->getTable_downgrade(MasterSegmentTable::EMTable);
      }
```
Obviously, the function signature would need to be changed appropriately to pass `emLocked` from the caller.

Reply:
Agree